distributed/tests/test_worker_state_machine.py

```diff
@@ -109,3 +111,17 @@ def test_sendmsg_to_dict():
     # Arbitrary sample class
     smsg = ReleaseWorkerDataMsg(key="x")
     assert smsg.to_dict() == {"op": "release-worker-data", "key": "x"}
+
+
+def test_merge_recs_instructions():
```

Comment:
What is the behaviour of `merge_recs_instructions` if a key in recommendations is duplicated?

Reply:
The newest one silently wins. I'm changing it to fail if one tries to change a previous recommendation with a new, non-identical one. This has a small performance cost which however is probably worth having.

Reply:
I don't care too much which way this goes but think we should be explicit about the behaviour. Being strict is fine but I've written code that deliberately overwrote a recommendation so I could see valid use cases in both directions.

If we get it working being strict, I'm happy